src/gltf/Raw2Gltf.cpp

```diff
@@ -261,41 +261,96 @@ ModelData* Raw2Gltf(
+            aoMetRoughTex = nullptr;
+            if (verboseOutput) {
+              fmt::printf("Material %s: no ORM textures detected\n", material.name.c_str() );
+            }
+          }
+          else if (isPassThroughTexture) {
+            // this handles the case where the same map is assigned to all the channels
             aoMetRoughTex = hasMetallicMap
                 ? simpleTex(RAW_TEXTURE_USAGE_METALLIC)
                 : (hasRoughnessMap
-                       ? simpleTex(RAW_TEXTURE_USAGE_ROUGHNESS)
-                       : (hasOcclusionMap ? simpleTex(RAW_TEXTURE_USAGE_OCCLUSION) : nullptr));
+                    ? simpleTex(RAW_TEXTURE_USAGE_ROUGHNESS)
+                    : (hasOcclusionMap
+                        ? simpleTex(RAW_TEXTURE_USAGE_OCCLUSION)
+                        : nullptr));
+            if (verboseOutput) {
+              if (aoMetRoughTex) {
+                fmt::printf("Material %s: detected single ORM texture: %s\n", material.name.c_str(), aoMetRoughTex->name.c_str());
+              } else {
+                fmt::printf("Material %s: no ORM textures detected\n", material.name.c_str() );
+              }
+            }
           } else {
-            // otherwise merge occlusion into the red channel, metallic into blue channel, and
-            // roughness into the green, of a new combinatory texture
+            /* otherwise we always have to create a new texture that merges
+             *   occlusion into the red channel
+             *   roughness into the green
+             *   metallic into blue channel
+             * with defaults for any unspecified channels
+             */
             aoMetRoughTex = textureBuilder.combine(
                 {
                     material.textures[RAW_TEXTURE_USAGE_OCCLUSION],
-                    material.textures[RAW_TEXTURE_USAGE_METALLIC],
                     material.textures[RAW_TEXTURE_USAGE_ROUGHNESS],
+                    material.textures[RAW_TEXTURE_USAGE_METALLIC],
                 },
                 "ao_met_rough",
                 [&](const std::vector<const TextureBuilder::pixel*> pixels)
                     -> TextureBuilder::pixel {
-                  const float occlusion = (*pixels[0])[0];
-                  const float metallic = (*pixels[1])[0] * (hasMetallicMap ? 1 : props->metallic);
-                  const float roughness =
-                      (*pixels[2])[0] * (hasRoughnessMap ? 1 : props->roughness);
+                  /**
+                   * note: we're picking the channels from the sources aligned with where they're going
+                   *   just in case they were authored that way. This makes an existing ORM texture
+                   *   "pass through", and has no effect on a grey single type texture.
+                   */
+                  const float occlusion = hasOcclusionMap ? (*pixels[0])[0] : 1;
+                  const float roughness = (*pixels[1])[1] * (hasRoughnessMap ? 1 : props->roughness);
```

Comment:
This is incorrect. It should be:
occlusion = pixels[0][0];
roughness = pixels[1][2];
metallic = pixels[2][1];

From GLTF spec:
The metalness values are sampled from the B channel. 
The roughness values are sampled from the G channel

https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/schema/material.pbrMetallicRoughness.schema.json

Reply:
Hey, thanks for engaging!

It's been a while, but if I recall correctly, what should be happening here is that `pixels` represents the 3 reads from the 3 input textures, with each pointing to a 3 value RGB array, either all being the same single packed ORM texture, or each being an individual greyscale texture, or some combination of both. In the PR as given, the occlusion looks for the red(0) value in texture 0, roughness the green(1) value in texture 1, and metalicness(2) the blue value in texture 2. I think the code you're suggesting would swizzle those reads?

Reply:
I think you are correct. My source texture I was using as reference had multiple versions, so I was comparing oranges to apples.  The tech artist had created one version of the PNG with the colors in the wrong channel & that is what I was using as my reference.

I actually folded your diff for detecting the same o,r,m textures & bypassing the pixel stuff, which worked for my case as well. Thank you!!!

Reply:
Glad to help : )